dynamic_rest/prefetch.py

```diff
@@ -41,6 +41,40 @@ def __getattr__(self, name):
             # Fast approach failed, fall back on slower logic.
             return self._slow_getattr(name)
 
+class SlowObject(dict):
+
+    def __init__(self, slow_object=None, *args, **kwargs):
+        self.pk_field = kwargs.pop('pk_field', 'id')
+        self.data = slow_object
+        return super(SlowObject, self).__init__(slow_object.__dict__, *args)
+
+    @property
+    def pk(self):
+        return self[self.pk_field]
+
+    def __getitem__(self, value):
+
+        if hasattr(self.data, str(value)):
+            return getattr(self.data, str(value))
+
+        # for the purposse of mapping serialized model + '_id' fields back to
+        # internal models, we need to check if that pattern is present
+        is_nested_obj = value.split('_')
```

Comment:
Can I get some explanation why this ends up being needed?  I'm not sure I follow the usage here.

Reply:
I can tell you why I did it, and my theory is [here](https://github.com/AltSchool/dynamic-rest/blob/2f744d150015341a6e4d6eca6c266373703208bd/dynamic_rest/serializers.py#L648)

One of the attributes requested was, I think, preview_of_id from the parent serializer because it's a related model.

That attribute doesn't exist on this model and so to return something I look to see if it follows that format and return the value it asks for.

Reply:
Ohh, interesting.  Yeah, this should work for now.  There's probably a more formal way to do it (using through fields or something), but this should work for now.  We'll run with it and see if any other things shake out.